homeassistant/components/econet/water_heater.py

```diff
@@ -1,153 +1,84 @@
 """Support for Rheem EcoNet water heaters."""
-import datetime
 import logging
 
-from pyeconet.api import PyEcoNet
-import voluptuous as vol
+from pyeconet.equipment import EquipmentType
+from pyeconet.equipment.water_heater import WaterHeaterOperationMode
 
 from homeassistant.components.water_heater import (
-    PLATFORM_SCHEMA,
+    ATTR_TEMPERATURE,
     STATE_ECO,
     STATE_ELECTRIC,
     STATE_GAS,
     STATE_HEAT_PUMP,
     STATE_HIGH_DEMAND,
     STATE_OFF,
     STATE_PERFORMANCE,
+    SUPPORT_AWAY_MODE,
     SUPPORT_OPERATION_MODE,
     SUPPORT_TARGET_TEMPERATURE,
     WaterHeaterEntity,
 )
-from homeassistant.const import (
-    ATTR_ENTITY_ID,
-    ATTR_TEMPERATURE,
-    CONF_PASSWORD,
-    CONF_USERNAME,
-    TEMP_FAHRENHEIT,
-)
-import homeassistant.helpers.config_validation as cv
+from homeassistant.const import TEMP_FAHRENHEIT
+from homeassistant.core import callback
 
-from .const import DOMAIN, SERVICE_ADD_VACATION, SERVICE_DELETE_VACATION
+from . import EcoNetEntity
+from .const import DOMAIN, EQUIPMENT
 
 _LOGGER = logging.getLogger(__name__)
 
-ATTR_VACATION_START = "next_vacation_start_date"
-ATTR_VACATION_END = "next_vacation_end_date"
 ATTR_ON_VACATION = "on_vacation"
-ATTR_TODAYS_ENERGY_USAGE = "todays_energy_usage"
 ATTR_IN_USE = "in_use"
 
-ATTR_START_DATE = "start_date"
-ATTR_END_DATE = "end_date"
-
-ATTR_LOWER_TEMP = "lower_temp"
-ATTR_UPPER_TEMP = "upper_temp"
-ATTR_IS_ENABLED = "is_enabled"
-
-SUPPORT_FLAGS_HEATER = SUPPORT_TARGET_TEMPERATURE | SUPPORT_OPERATION_MODE
-
-ADD_VACATION_SCHEMA = vol.Schema(
-    {
-        vol.Optional(ATTR_ENTITY_ID): cv.entity_ids,
-        vol.Optional(ATTR_START_DATE): cv.positive_int,
-        vol.Required(ATTR_END_DATE): cv.positive_int,
-    }
-)
-
-DELETE_VACATION_SCHEMA = vol.Schema({vol.Optional(ATTR_ENTITY_ID): cv.entity_ids})
-
-ECONET_DATA = "econet"
-
 ECONET_STATE_TO_HA = {
-    "Energy Saver": STATE_ECO,
-    "gas": STATE_GAS,
-    "High Demand": STATE_HIGH_DEMAND,
-    "Off": STATE_OFF,
-    "Performance": STATE_PERFORMANCE,
-    "Heat Pump Only": STATE_HEAT_PUMP,
-    "Electric-Only": STATE_ELECTRIC,
-    "Electric": STATE_ELECTRIC,
-    "Heat Pump": STATE_HEAT_PUMP,
+    WaterHeaterOperationMode.ENERGY_SAVING: STATE_ECO,
+    WaterHeaterOperationMode.HIGH_DEMAND: STATE_HIGH_DEMAND,
+    WaterHeaterOperationMode.OFF: STATE_OFF,
+    WaterHeaterOperationMode.HEAT_PUMP_ONLY: STATE_HEAT_PUMP,
+    WaterHeaterOperationMode.ELECTRIC_MODE: STATE_ELECTRIC,
+    WaterHeaterOperationMode.GAS: STATE_GAS,
+    WaterHeaterOperationMode.PERFORMANCE: STATE_PERFORMANCE,
 }
+HA_STATE_TO_ECONET = {value: key for key, value in ECONET_STATE_TO_HA.items()}
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
```

Comment:
Keep the platform schema until we remove the config import. Log a warning that it's deprecated.